### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by y0rrrsh**
Tab bar navigation

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -43,3 +43,8 @@ Carthage/Build
 # fastlane
 fastlane/report.xml
 fastlane/screenshots
+.idea/misc.xml
+.idea/modules.xml
+.idea/VkAudio.iml
+.idea/workspace.xml
+.idea/xcode.xml
```
Comment (taviscaron) [inline .gitignore]: why not ignore .idea/ ?

File: VkAudio/cells/FriendCell.swift [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+class FriendCell: UITableViewCell {
+
+    static let STORYBOARD_ID = "cell_friend"
+
+    @IBOutlet weak var imagePhoto: UIImageView!
+    @IBOutlet weak var labelName: UILabel!
+
+    func update(firstName: String, lastName: String, photoUrl: String) {
+        labelName.text = "\(firstName) \(lastName)"
+        imagePhoto.kf_setImageWithURL(NSURL(string: photoUrl)!)
```
Comment (taviscaron) [inline VkAudio/cells/FriendCell.swift]: crash if url is malformed

File: VkAudio/cells/GroupCell.swift [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+class GroupCell: UICollectionViewCell{
+    
+    static let STORYBOARD_ID = "cell_group"
+    
+    @IBOutlet weak var imagePhoto: UIImageView!
+    @IBOutlet weak var labelName: UILabel!
+    
+    func update(groupName: String, photoUrl: String){
+        imagePhoto.kf_setImageWithURL(NSURL(string: photoUrl)!)
```
Comment (taviscaron) [inline VkAudio/cells/GroupCell.swift]: same as above: possible crash

File: VkAudio/viewcontroller/UserAudiosViewController.swift [hunk 4]
```diff
@@ -2,23 +2,51 @@
...
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        VkSDK.instance?.getAudios({ (result) in
-            print(result)
+        VkSDK.Users.getUserInfo(
+            ["fields" : "photo_big"],
+            onResult: { result in
+                let url = NSURL(string: result["photo_big"] as! String)
```
Comment (taviscaron) [inline VkAudio/viewcontroller/UserAudiosViewController.swift]: possible crash

File: VkAudio/viewcontroller/UserAudiosViewController.swift [hunk 5]
```diff
@@ -2,23 +2,51 @@
...
+                let imageView = UIImageView()
+                imageView.kf_setImageWithURL(url!)
```
Comment (taviscaron) [inline VkAudio/viewcontroller/UserAudiosViewController.swift]: crash ?

File: VkAudio/viewcontroller/UserFriendsViewController.swift [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import Foundation
+import UIKit
+
+class UserFriendsViewController: UITableViewController {
+    
+    var friends = [Friend]()
+
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+        VkSDK.Frinds.getFriends(["fields": "name, photo_50"], onResult: { (result) in
```
Comment (taviscaron) [inline VkAudio/viewcontroller/UserFriendsViewController.swift]: api-related stuff on UI. should be encapsulated into service level

File: VkAudio/viewcontroller/UserFriendsViewController.swift [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return friends.count
+    }
+    
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithIdentifier(FriendCell.STORYBOARD_ID) as! FriendCell
+        let friend: Friend = friends[indexPath.row]
+        
+        cell.backgroundColor = UIColor(white: 1, alpha: 0.5)
+        cell.update(friend.firstName!, lastName: friend.lastName!, photoUrl: friend.photoUrl!)
```
Comment (taviscaron) [inline VkAudio/viewcontroller/UserFriendsViewController.swift]: crash crash crash

File: VkAudio/vk/Audios.swift [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    struct Audios {
+        
+        static func getAudios(params: [String: AnyObject]? = nil, onResult: (result: [Audio]) -> Void, onError: ((error: NSError) -> Void)? = {error in print(error)}){
+            VkSDK.instance?.get("audio.get", parameters: params, onResult: { (result) in
+                var audios = [Audio]()
+                for i in 1 ..< result.count {
+                    let audio = Audio(apiResponse: result[i] as! [String: AnyObject])
+                    audios.append(audio)
+                }
+                onResult(result: audios)
```
Comment (taviscaron) [inline VkAudio/vk/Audios.swift]:
```
audios = result.flatMap { result in
    if let result = result as? [String: AnyObject] {
        return Audio(apiResponse: result)
    } else {
        return nil
   }
}
```

File: VkAudio/vk/VkSDK.swift [hunk 9]
```diff
@@ -53,7 +53,35 @@ class VkSDK {
...
+            .responseJSON{ apiResponse in
+                if let response = apiResponse.result.value {
+                    let responseDict = response as! [String : AnyObject]
+                    if let responseDictValues = responseDict["response"] as? [AnyObject]{
+                        onResult(result: Array(responseDictValues))
+                    }
+                    else if let errorDictValues = responseDict["error"] {
+                        let code = errorDictValues["error_code"] as! Int
+                        let message = errorDictValues["error_msg"] as! String
+
```
Comment (taviscaron) [inline VkAudio/vk/VkSDK.swift]: crash =)

File: .gitignore (hunk 1 above)
Comment (y0rrrsh) [inline .gitignore]: generated by source tree. will change =)

File: VkAudio/vk/Audios.swift (hunk 8 above)
Comment (taviscaron) [inline VkAudio/vk/Audios.swift]: or even that:
```
audios = result.flatMap { $0 as [String: AnyObject] }.map { Audio(apiResponse: $0) }
```

File: VkAudio/viewcontroller/UserFriendsViewController.swift (hunk 6 above)
Comment (y0rrrsh) [inline VkAudio/viewcontroller/UserFriendsViewController.swift]: will fix while refactoring

File: VkAudio/vk/VkSDK.swift (hunk 9 above)
Comment (y0rrrsh) [inline VkAudio/vk/VkSDK.swift]: ...on error handling

**Pull request merged**